internal/provisioner/kops_provisioner_cluster.go

```diff
@@ -302,24 +304,34 @@ func (provisioner *KopsProvisioner) ProvisionCluster(cluster *model.Cluster, aws
 		return errors.Wrap(err, "failed to delete PodDisruptionBudget calico-typha")
 	}
 
-	// Need to remove two items from the metrics-server because the fields after the creation are imutable so the
+    cleanupMetricsServerResources(provisioner, cluster)
+
+	// Need to remove 3 items from the prometheus-adapter because the fields after the creation are immutable so the
 	// create/update does not work. We might want to refactor this in the future to avoid this
-	logger.Info("Cleaning up some metrics-server resources to reapply")
-	err = k8sClient.Clientset.CoreV1().Services("kube-system").Delete(ctx, "metrics-server", metav1.DeleteOptions{})
+	logger.Info("Cleaning up some prometheus-adapter resources to reapply")
+	err = k8sClient.Clientset.CoreV1().Services("prometheus").Delete(ctx, "prometheus-adapter", metav1.DeleteOptions{})
 	if k8sErrors.IsNotFound(err) {
-		logger.Info("Service metrics-server not found; skipping...")
+		logger.Info("Service prometheus-adapter not found; skipping...")
 	} else if err != nil {
 		return errors.Wrap(err, "failed to delete service metrics-server")
 	}
 
-	logger.Info("Cleaning up some metrics-server resources to reapply")
+	logger.Info("Cleaning up some prometheus-adapter resources to reapply")
 	err = k8sClient.KubeagClientSet.ApiregistrationV1beta1().APIServices().Delete(ctx, "v1beta1.metrics.k8s.io", metav1.DeleteOptions{})
 	if k8sErrors.IsNotFound(err) {
 		logger.Info("APIService v1beta1.metrics.k8s.io not found; skipping...")
 	} else if err != nil {
 		return errors.Wrap(err, "failed to delete APIService v1beta1.metrics.k8s.io")
 	}
 
+	logger.Info("Cleaning up some prometheus-adapter resources to reapply")
```

Comment:
Another duplicate log message. I would just print this once before we begin cleanup.